task6/Working with posts.js

```diff
@@ -0,0 +1,251 @@
+new post("29", "Ava.jpg", "incredible4",  new Date('2020-07-05T20:30:55'), "Hello931"),
+new post("30", "Ava.jpg", "incredible4",  new Date('2020-04-25T00:08:16'), "Hello990"),
+new post("31", "Ava.jpg", "incredible5",  new Date('2020-07-10T09:20:03'), "Hello633"),
+new post("32", "Ava.jpg", "incredible5",  new Date('2020-09-27T18:55:54'), "Hello468"),
+new post("33", "Ava.jpg", "incredible5",  new Date('2020-12-27T12:39:01'), "Hello530", ["#hai"]),
+new post("34", "Ava.jpg", "incredible5",  new Date('2020-12-20T18:54:23'), "Hello927", ["#ni hao"]),
+new post("35", "Ava.jpg", "incredible5",  new Date('2020-07-19T11:25:13'), "Hello549"),
+new post("36", "Ava.jpg", "incredible5",  new Date('2020-10-26T40:03:24'), "Hello616"),
+new post("37", "Ava.jpg", "incredible5",  new Date('2020-12-23T17:20:30'), "Hello682"),
+new post("38", "Ava.jpg", "incredible5",  new Date('2020-07-28T00:46:26'), "Hello208"),
+new post("39", "Ava.jpg", "incredible5",  new Date('2020-10-22T02:59:26'), "Hello940"),
+new post("40", "Ava.jpg", "incredible5",  new Date('2020-03-15T21:40:13'), "Hello999")
+];
+
+
+class editInfo{
+	constructor(description, hashTags = [], photoLink="") {
+		this.description = description;
+		this.hashTags = hashTags;
+		this.photoLink = photoLink;
+	}
+}
+
+class filterParams {
+	constructor (author, date, hashTags=[]) {
+		this.author = author;
+		this.date = date;
+		this.hashTags = hashTags;
+	}
+}
+
+function inOneDay(date1, date2) {
+	return date1.getFullYear() == date2.getFullYear()
+		 && date1.getMonth() == date2.getMonth()
+		 && date1.getDate() == date2.getDate();
+}
+
+function hasAnyTag(hashTagsOfPost, hashTags) {
+	return hashTagsOfPost.some(hashTagOfPost=>(hashTags.some(hashTag=>hashTag==hashTagOfPost)));
+}
+
+function getPostsByFilter(filterConfig) {
+	return posts.filter(element=>
+		(element.author == filterConfig.author) &&
+		((filterConfig.date == null) || inOneDay(element.createdAt, filterConfig.date)) &&
+		 ((filterConfig.hashTags == []) || hasAnyTag(element.hashTags, filterConfig.hashTags))
+		 );
+}
+
+function compDates(x, y) {
+	return x.createdAt < y.createdAt ? -1 : x.createdAt > y.createdAt ? 1 : 0;
+}
+
+function getPosts(filterConfig, skip=0, top=10) {
+	if (skip < 0 || top <= 0) {
+		return [];
+	}
+	let postsByFilter = getPostsByFilter(filterConfig).sort(compDates);
+	return postsByFilter.slice(skip, skip + top);
+}
+
+function getPost(ID) {
+	return posts.find(function(item, index, array) {
+		return item.id == ID;
+	});
+}
+
+function validatePost(somePost) {
+	return somePost.description.length > 0 && somePost.description.length < 200 &&
+	 somePost.author.length > 0;
+}
+
+function addPost(somePost) {
+	if (!validatePost(somePost)) {
+		return false;
+	}
+	posts.push(somePost);
+	return true;
+}
+
+function editPost(ID, edit) {
+	let edPost = getPost(ID);
+	if (edPost==undefined) {
+		return false;
+	}
+	if (!validatePost(edPost)) {
+		return false;
+	}
+	let index = posts.indexOf(edPost);
```

Comment:
You are iterating over array twice. First time when finding the post, now for finding it's index. Can `findindex` be used to find index and the just take post from array by index? https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex